polkadot/node/service/src/lib.rs

```diff
@@ -944,14 +944,9 @@ pub fn new_full<
 				secure_validator_mode,
 				prep_worker_path,
 				exec_worker_path,
-				pvf_execute_workers_max_num: execute_workers_max_num.unwrap_or_else(
-					|| match config.chain_spec.identify_chain() {
-						// The intention is to use this logic for gradual increasing from 2 to 4
-						// of this configuration chain by chain until it reaches production chain.
-						Chain::Polkadot | Chain::Kusama => 2,
-						Chain::Rococo | Chain::Westend | Chain::Unknown => 4,
-					},
-				),
+				// Default execution workers is 4 because we have 8 cores on the reference hardware,
+				// and this accounts for 50% of that cpu capacity.
+				pvf_execute_workers_max_num: execute_workers_max_num.unwrap_or(4),
```

Comment:
Do we maybe want to limit this to Kusama first?

Reply:
I'm fine both ways, what do you think @sandreim  ?

Reply:
I'm not aware of any risks from this change. I would go for both. Exceptionally, If something bag happens we have CLI param to fix it back.

Reply:
Yeah, I thought about it as well and both make more sense, since people should have already started to use the new reference hw, so there is no point not to try to use the available capacity.